File: samples/snippets/visualbasic/VS_Snippets_Winforms/Classic DataObject.SetData2 Example/VB/source.vb
﻿Imports System.Data
Imports System.ComponentModel
Imports System.Windows.Forms



Public Class Form1
    Inherits Form
    Protected textBox1 As TextBox
    
    ' <Snippet1>
    Private Sub AddMyData2()
        ' Creates a component to store in the data object.
        Dim myComponent As New Component()
        
        ' Gets the type of the component.
        Dim myType As Type = myComponent.GetType()
        
        ' Creates a new data object.
        Dim myDataObject As New DataObject()
        
        ' Adds the component to the DataObject.
        myDataObject.SetData(myType, myComponent)
        
        ' Prints whether data of the specified type is in the DataObject.
        If myDataObject.GetDataPresent(myType) Then
            textBox1.Text = "Data of type " & myType.GetType().Name & _
                " is present in the DataObject"
        Else
            textBox1.Text = "Data of type " & myType.GetType().Name & _
                " is not present in the DataObject"
        End If
    End Sub
    ' </Snippet1>
End Class

File: samples/snippets/csharp/VS_Snippets_Winforms/Classic RecommendedAsConfigurableAttribute.RecommendedAsConfigurableAttribute Example/CS/source.cs
﻿using System;
using System.Data;
using System.ComponentModel;
using System.Windows.Forms;

public class Form1: Form
{
 protected TextBox textBox1;
 
// <Snippet1>
 [RecommendedAsConfigurable(true)]
 public int MyProperty {
    get {
       // Insert code here.
       return 0;
    }
    set {
       // Insert code here.
    }
 }
    
// </Snippet1>
}
